languages/rust/oso/src/host/class.rs

```diff
@@ -130,6 +131,31 @@ impl<T> Class<T> {
         self
     }
 
+    pub fn add_result_method<F, Args, R, E>(mut self, name: &str, f: F) -> Self
+    where
+        Args: FromPolar,
+        F: Method<T, Args, Result = Result<R, E>> + 'static,
+        R: ToPolar + 'static,
+        E: Debug + 'static,
```

Comment:
This would be better if it just relied on `E: std::error::Error`, and then used `err.to_string()` later.